streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedStreamsTasks.java

```diff
@@ -89,14 +88,12 @@ int maybeCommit() {
     int process() {
         int processed = 0;
 
-        // only iterate over tasks who has data in all its input topic partitions
-        final Map<TaskId, StreamTask> processable = running.entrySet().stream()
-                .filter(entry -> entry.getValue().isProcessable())
-                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        final Iterator<Map.Entry<TaskId, StreamTask>> it = running.entrySet().iterator();
+        while (it.hasNext()) {
+            final StreamTask task = it.next().getValue();
```

Comment:
Originally I was ok with the Java 8 approach, but this it bit better in that we only traverse tasks once.